From 4ed232cead80b94a3ff7220ec8bc4b7c7979917d Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Thu, 21 May 2020 20:41:58 +0800
Subject: [PATCH] usb: dwc3: core: increase the DWC3_ISOC_MAX_RETRIES to 50

If usb gadget isochronous run in a high load and high bus latency
system, it may fail to start a transfer for isochronous endpoint,
and retry 5 times is not enough. This patch increases the retry
times to 50.

Change-Id: Id95a2b4fb09c3103a1da456b6ccb54ed24d03197
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/dwc3/core.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/core.h b/drivers/usb/dwc3/core.h
index 871756f4ebec..8686cab787e1 100644
--- a/drivers/usb/dwc3/core.h
+++ b/drivers/usb/dwc3/core.h
@@ -37,7 +37,7 @@
 #define DWC3_EP0_SETUP_SIZE	512
 #define DWC3_ENDPOINTS_NUM	32
 #define DWC3_XHCI_RESOURCES_NUM	2
-#define DWC3_ISOC_MAX_RETRIES	5
+#define DWC3_ISOC_MAX_RETRIES	50
 
 #define DWC3_SCRATCHBUF_SIZE	4096	/* each buffer is assumed to be 4KiB */
 #define DWC3_EVENT_BUFFERS_SIZE	4096
-- 
2.35.3

